### PR TITLE
En/websocket write message

### PR DESCRIPTION
**Pull request by Umang01-hash**
## Pull Request Template

**Description:**

-   Resolves issue #881 
-   Introduce a new `WriteMessageToSocket ` method in context for writing message to socket.

### Example Usage

```go
func WSHandler(ctx *gofr.Context) (interface{}, error) {
	var message string

	err := ctx.Bind(&message)
	if err != nil {
		ctx.Logger.Errorf("Error binding message: %v", err)
		return nil, err
	}

	ctx.Logger.Infof("Received message: %s", message)

	err = ctx.WriteMessageToSocket("this is new message")
	if err != nil {
		return nil, err
	}

	return message, nil
}
```

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [ ] All new code is covered by unit tests.
-   [ ] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

**Review by ccoVeille — APPROVED** [review]

File: pkg/gofr/context.go [hunk 1]
```diff
@@ -58,6 +61,33 @@ func (c *Context) Bind(i interface{}) error {
 	return c.Request.Bind(i)
 }
 
+// WriteMessageToSocket writes a message to the WebSocket connection associated with the context.
+// The data parameter can be of type string, []byte, or any struct that can be marshaled to JSON.
+// It retrieves the WebSocket connection from the context and sends the message as a TextMessage.
+func (c *Context) WriteMessageToSocket(data interface{}) error {
+	// Retrieve connection from context based on connectionID
+	conn := c.GetConnectionFromContext(c)
```
Comment (vipul-rawat) [inline pkg/gofr/context.go:69]: This is confusion although syntactically they are correct.
```suggestion
	conn := c.Container.GetConnectionFromContext(c)
```

or 

```suggestion
    conn := c.GetConnectionFromContext(c.Context)
```

Comment (Umang01-hash) [reply pkg/gofr/context.go]: done

File: pkg/gofr/websocket.go [hunk 2]
```diff
@@ -45,9 +52,36 @@ func handleWebSocketConnection(ctx *Context, conn *websocket.Connection, handler
...
+		}
+
+		err = conn.WriteMessage(websocket.TextMessage, message)
 		if err != nil {
 			ctx.Errorf("Error writing message: %v", err)
 		}
 	}
 }
+
+func serializeMessage(response interface{}) ([]byte, error) {
```
Comment (vipul-rawat) [inline pkg/gofr/websocket.go:68]:
```suggestion
func serializeMessage(response any) ([]byte, error) {
```

File: pkg/gofr/context.go [hunk 3]
```diff
@@ -58,6 +60,21 @@ func (c *Context) Bind(i interface{}) error {
 	return c.Request.Bind(i)
 }
 
+// WriteMessageToSocket writes a message to the WebSocket connection associated with the context.
+// The data parameter can be of type string, []byte, or any struct that can be marshaled to JSON.
+// It retrieves the WebSocket connection from the context and sends the message as a TextMessage.
+func (c *Context) WriteMessageToSocket(data interface{}) error {
```
Comment (vipul-rawat) [inline pkg/gofr/context.go:66]:
```suggestion
func (c *Context) WriteMessageToSocket(data any) error {
```

Comment (Umang01-hash) [reply pkg/gofr/context.go]: done

File: pkg/gofr/websocket.go (hunk 2 above)
Comment (Umang01-hash) [reply pkg/gofr/websocket.go]: done

**Review by vipul-rawat — APPROVED** [review]

**Review by aryanmehrotra — APPROVED** [review]

**Pull request merged**